### PR TITLE
Add strict arg to Prompt.build

### PR DESCRIPTION
**Pull request by dpguthrie**
Adding a `strict` argument to the Prompt.build method.  When `True` it will throw an error for missing variables.

This functionality exists in the typescript sdk:

- https://github.com/braintrustdata/braintrust-sdk/blob/3d5ec4de453ab8ea873a6c8c0d0ef0b2110bd386/js/src/logger.ts#L6012
- https://github.com/braintrustdata/braintrust-sdk/blob/main/js/src/mustache-utils.ts

### CODE REVIEW

**Comment by dpguthrie** [comment]
@ankrgyl Added the following:

Helper Function Tests (6 tests)
  - Mustache variable extraction (_extract_mustache_variables)
  - Nested value access (_get_nested_value)
  - Template validation (lint_template)

Integration Tests (7 tests)
  - Strict mode enabled: Validates missing variables throw ValueError
  with descriptive messages
  - Strict mode disabled: Ensures backward compatibility - missing
  variables render as empty strings, no exceptions
  - Nested objects: Tests dot notation like user.profile.email
  - Array access: Validates array bounds checking for items.0, items.1,
  etc.
  - Edge cases: Empty arrays, partial variables, completely missing
  variables

Key Behaviors Verified
  - strict=True: Missing variables → ValueError("Variable 'name' does 
  not exist.")
  - strict=False (default): Missing variables → empty strings, always
  generates valid prompt
  - Backward compatibility: Existing code continues to work unchanged
  - Error messages use original variable names (not normalized array
  indices)

File: py/src/braintrust/logger.py [hunk 1]
```diff
@@ -4291,6 +4297,27 @@ def render_prompt_params(params: Dict[str, Any], args: Any) -> Dict[str, Any]:
...
+def render_mustache(template: str, data: Any, *, strict: bool = False, renderer: Optional[Callable[..., Any]] = None):
+    if renderer is None:
+        renderer = chevron.render
+
+    if not strict:
+        return renderer(template, data=data)
+
+    # Capture stderr to check for missing keys
+    stderr_capture = io.StringIO()
+    with contextlib.redirect_stderr(stderr_capture):
```
Comment (ibolmo) [inline py/src/braintrust/logger.py:4309]: opted with this to defer as much as we can to the actual template rendering library
